src/app/d2-vendors/vendor-ratings.ts

```diff
@@ -0,0 +1,41 @@
+import { DestinyTrackerServiceType } from "../item-review/destiny-tracker.service";
+import { DestinyVendorsResponse, DestinyVendorSaleItemComponent, DestinyVendorResponse, DestinyProfileResponse, DestinyVendorItemDefinition } from "bungie-api-ts/destiny2";
+import * as _ from "underscore";
+import { D2ManifestDefinitions } from "../destiny2/d2-definitions.service";
+
+export async function fetchRatings(destinyTrackerService: DestinyTrackerServiceType,
+                                   vendorsResponse?: DestinyVendorsResponse,
+                                   vendorResponse?: DestinyVendorResponse,
+                                   profileResponse?: DestinyProfileResponse,
+                                   defs?: D2ManifestDefinitions): Promise<void> {
+  if (vendorsResponse) {
+    const saleComponentArray = Object.values(vendorsResponse.sales.data)
+      .map((saleItemComponent) => saleItemComponent.saleItems);
+
+    const saleComponents = ([] as DestinyVendorSaleItemComponent[]).concat(...saleComponentArray.map((v) => Object.values(v)));
```

Comment:
Hmm, I'm not sure of the value of concating onto an empty array. Why not just map the items? Or is this like `flatMap`?

Reply:
Maybe `flatMap` has better luck dealing with values extracted from a dictionary than _.flatten/Array.flatten() does?